pkg/resources/configmaps.go

```diff
@@ -0,0 +1,325 @@
+            ca_file /fluentd/etc/tls/qradar.crt
+            packet_size 4096
+            program fluentd
+            <format>
+                @type single_value
+                message_key message
+            </format>
+        </store>`
+var qRadarConfigData2 = `
+    </match>`
+
+// DataSplunk defines the struct for splunk-hec-config
+type DataSplunk struct {
+	Value string `yaml:"splunkHEC.conf"`
+}
+
+// DataQRadar defines the struct for remote-syslog-config
+type DataQRadar struct {
+	Value string `yaml:"remoteSyslog.conf"`
+}
+
+// BuildConfigMap returns a ConfigMap object
+func BuildConfigMap(instance *operatorv1alpha1.AuditLogging, name string) (*corev1.ConfigMap, error) {
+	reqLogger := log.WithValues("ConfigMap.Namespace", InstanceNamespace, "ConfigMap.Name", name)
+	metaLabels := LabelsForMetadata(FluentdName)
+	dataMap := make(map[string]string)
+	var err error
+	switch name {
+	case FluentdDaemonSetName + "-" + ConfigName:
+		dataMap[enableAuditLogForwardKey] = strconv.FormatBool(instance.Spec.Fluentd.EnableAuditLoggingForwarding)
+		type Data struct {
+			Value string `yaml:"fluent.conf"`
+		}
+		d := Data{}
+		err = yaml.Unmarshal([]byte(fluentdMainConfigData), &d)
+		dataMap[fluentdConfigKey] = d.Value
+	case FluentdDaemonSetName + "-" + SourceConfigName:
+		type DataS struct {
+			Value string `yaml:"source.conf"`
+		}
+		ds := DataS{}
+		var result string
+		if instance.Spec.Fluentd.JournalPath != "" {
+			result = sourceConfigData1 + instance.Spec.Fluentd.JournalPath + sourceConfigData2
+		} else {
+			result = sourceConfigData1 + defaultJournalPath + sourceConfigData2
+		}
+		p := strconv.Itoa(defaultHTTPPort)
+		result += sourceConfigData3 + p + sourceConfigData4
+		err = yaml.Unmarshal([]byte(result), &ds)
+		dataMap[SourceConfigKey] = ds.Value
+	case FluentdDaemonSetName + "-" + SplunkConfigName:
+		dsplunk := DataSplunk{}
+		err = yaml.Unmarshal([]byte(splunkConfigData1+splunkDefaults+splunkConfigData2), &dsplunk)
+		if err != nil {
+			reqLogger.Error(err, "Failed to unmarshall data for "+name)
+		}
+		dataMap[SplunkConfigKey] = dsplunk.Value
+	case FluentdDaemonSetName + "-" + QRadarConfigName:
+		dq := DataQRadar{}
+		err = yaml.Unmarshal([]byte(qRadarConfigData1+qRadarDefaults+qRadarConfigData2), &dq)
+		dataMap[QRadarConfigKey] = dq.Value
+	default:
+		reqLogger.Info("Unknown ConfigMap name")
+	}
+	if err != nil {
+		reqLogger.Error(err, "Failed to unmarshall data for "+name)
+		return nil, err
+	}
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: InstanceNamespace,
+			Labels:    metaLabels,
+		},
+		Data: dataMap,
+	}
+	return cm, nil
+}
+
+func getConfig(data string) (string, error) {
+	var siemConfig string
+	// >([^]]+)< retrieves all data between match tags
+	regex := `>\n([^]]+)\n<`
+	reConfig := regexp.MustCompile(regex)
+	matches := reConfig.FindStringSubmatch(data)
+	if len(matches) < 2 {
+		return "", errors.New("output plugin config misformatted")
```

Comment:
Do you want to add the incorrect value to the log output?

Reply:
updated